src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const arrayWithObjects = [];
+  let cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        cloneState = {
+          ...cloneState,
+          ...action.extraData,
+        };
+
+        break;
+
+      case 'removeProperties':
+        cloneState = removeProperties(cloneState, action.keysToRemove);
+
+        break;
+      case 'clear':
+        cloneState = {};
+
+        break;
+    }
+    arrayWithObjects.push({ ...cloneState });
+  }
+
+  return arrayWithObjects;
+}
+
+function removeProperties(newObject, keysToRemove) {
+  for (const key of keysToRemove) {
+    delete newObject[key];
+  }
+
+  return newObject;
 }
```

Comment:
[NAMING] - use proper names for object copy. In the `removeProperties` function, the parameter `newObject` should be named to reflect that it is a clone of the state, such as `stateClone`.